src/policy/immix/immixspace.rs

```diff
@@ -113,6 +117,46 @@ impl<VM: VMBinding> Space<VM> for ImmixSpace<VM> {
     }
 }
 
+impl<VM: VMBinding> crate::policy::gc_work::PolicyTraceObject<VM> for ImmixSpace<VM> {
+    #[inline(always)]
+    fn trace_object<T: TransitiveClosure, const KIND: crate::policy::gc_work::TraceKind>(
+        &self,
+        trace: &mut T,
+        object: ObjectReference,
+        copy: Option<CopySemantics>,
+        worker: &mut GCWorker<VM>,
+    ) -> ObjectReference {
+        if KIND == TRACE_KIND_DEFRAG {
+            self.trace_object(trace, object, copy.unwrap(), worker)
+        } else if KIND == TRACE_KIND_FAST {
+            self.fast_trace_object(trace, object)
+        } else {
+            unreachable!()
+        }
+    }
+
+    #[inline(always)]
+    fn create_scan_work<E: ProcessEdgesWork<VM = VM>>(
+        &'static self,
+        nodes: Vec<ObjectReference>,
+    ) -> Box<dyn GCWork<VM>> {
+        Box::new(crate::policy::immix::ScanObjectsAndMarkLines::<E>::new(
+            nodes, false, self,
+        ))
+    }
```

Comment:
I have some crazy ideas about removing ScanObjectsAndMarkLines completely.

https://github.com/mmtk/mmtk-core/issues/576

You can try it in this PR if it can make your work less laborious, or we can do it in another PR.

Reply:
This change is done in the PR.